File: Backend/BankingPaymentsApp_API/BankingPaymentsApp_API/DTOs/SalaryResponseDTO.cs
using BankingPaymentsApp_API.Models;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BankingPaymentsApp_API.DTOs
{
    public class SalaryResponseDTO
    {
        public int SalaryDisbursementId { get; set; }
        public int ClientId { get; set; }
        public decimal TotalAmount { get; set; }
        public DateTime DisbursementDate { get; set; } = DateTime.UtcNow;
        public int DisbursementStatusId { get; set; } = 3;
        public bool AllEmployees { get; set; } = true;
        public virtual ICollection<SalaryDisbursementDetails>? DisbursementDetails { get; set; } = new List<SalaryDisbursementDetails>();
    }
}

File: frontend/BankingPaymentsApp_Frontend/src/DTO/SalaryResponseDTO.ts
import { SalaryDisbursementDetails } from "../Models/SalaryDisbursementDetails";

export interface SalaryResponseDTO {
  SalaryDisbursementId: number;
  ClientId: number;
  TotalAmount: number; 
  DisbursementDate: string; 
  DisbursementStatusId: number;
  AllEmployees: boolean;
  DisbursementDetails?: Array<SalaryDisbursementDetails>; 
}
